### DISCUSSION TITLE
documentation patches...

### DISCUSSION THREAD

**Discussion by thebjorn** [nextauthjs/next-auth#10106] [Help]
Is there any guides on how to build the documentation?

I would like to help with the documentation, but I'm not too familiar with how documentation is done in the js world. There is a "PRs welcome" link directly to the github file, but I'm assuming you want us to fork/fix/build-docs before submitting a PR?

In particular, I'm looking at this page (https://authjs.dev/reference/sveltekit) where there is a mix of tabs/spaces in some of the code examples making the indentation "unsynchronized" ;-)   Some of the examples indicate which filename they're from, while some don't, which makes it a bit confusing. Code that is mentioned should go in server files don't always have examples. All of these should be easy to fix.

Svelte 5 is not released yet, but many of us are building future projects with it already. Would there be any interest in adding Svelte 5 versions of the examples?

**Comment by programm-ingovals** [comment]
I'm also looking for a discussion on the docs on the site.

This site https://next-auth.js.org/getting-started/example#add-api-route has a green tip box
> If you are using the Next.js App Router, please note that <SessionProvider /> requires a client component and therefore cannot be put inside the root layout. For more details, check out the [Next.js documentation](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts).

But if I understand the [Next.js docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#using-context-providers) correctly this isn't really an issue and you can put SessionProvider straight into a ServerComponent since it wraps the actually hook provider in a 'use client' component.
